src/main/java/pl/agh/edu/iisg/io/vmms/vmmsbackend/controller/UserController.java

```diff
@@ -28,34 +34,43 @@ public UserController(UserService userService) {
         this.userService = userService;
     }
 
-    @RequestMapping("/")
+    @RequestMapping(MAIN_ENDPOINT)
     public String index() {
         return "Virtual Machine Management System";
     }
 
-    @RequestMapping(path = "/users", method = RequestMethod.GET)
-    public List<User> getUsers() {
-        return userService.getUsers();
+    @RequestMapping(path = USERS_ENDPOINT, method = RequestMethod.GET)
+    public List<UserDto> getUsers() {
+        return userService.getUsers().stream()
+                .map(user -> new UserDto(user.getId(), user.getUserName(), user.isAdmin()))
+                .collect(Collectors.toList());
     }
 
-    @RequestMapping(path = "/users/{name}", method = RequestMethod.GET)
-    public List<User> getUsers(@PathVariable String name) {
-        return userService.findByNameContaining(name);
+    @RequestMapping(path = USERS_ENDPOINT + "/{name}", method = RequestMethod.GET)
+    public List<UserDto> getUsers(@PathVariable String name) {
+        return userService.findByNameContaining(name).stream()
+                .map(user -> new UserDto(user.getId(), user.getUserName(), user.isAdmin()))
+                .collect(Collectors.toList());
     }
 
-    @RequestMapping(path = "/user/create", method = RequestMethod.POST)
+    @RequestMapping(path = CREATE_USER_ENDPOINT, method = RequestMethod.POST)
     @ResponseStatus(HttpStatus.CREATED)
-    public User createUser(@RequestBody Map<String, String> request) throws HttpException {
-        String userName = Optional.ofNullable(request.get(NAME)).orElseThrow(MissingUserNameException::new);
-        return userService.save(userName);
+    public UserDto createUser(@RequestBody Map<String, Object> request) throws HttpException {
+        String userName = String.valueOf(Optional.ofNullable(request.get(USER_NAME)).orElseThrow(MissingUserNameException::new));
+        Boolean isAdmin = (Boolean) Optional.ofNullable(request.get(USER_ADMIN)).orElse(false);
+        User user = userService.save(userName, isAdmin);
+        return new UserDto(user.getId(), user.getUserName(), user.isAdmin());
     }
 
-    @RequestMapping(path = "/user/delete", method = RequestMethod.POST)
+    @RequestMapping(path = DELETE_USER_ENDPOINT, method = RequestMethod.POST)
     @ResponseStatus(HttpStatus.OK)
-    public User deleteUser(@RequestBody Map<String, Long> request) throws HttpException {
-        Long userId = Optional.ofNullable(request.get(USER_ID)).orElseThrow(MissingUserNameException::new);
+    public UserDto deleteUser(@RequestBody Map<String, Object> request) throws HttpException {
```

Comment:
I don't get the necessity for the request map here :/ Why this couldn't be a simple object?